docs/content/dagster-cloud/developing-testing/code-locations.mdx

```diff
@@ -1,7 +1,245 @@
+
+---
+
+## Dagster Cloud code requirements
+
+To work with Dagster Cloud, your Dagster code:
+
+- **Must be loaded from a single entry point, either a Python file or package.** This entry point can load repositories from other files or packages.
+
+- **Must run in an environment where the `dagster` and `dagster-cloud` 0.13.2 or later Python packages are installed.**
+
+- **If using a [Hybrid deployment](/dagster-cloud/deployment/hybrid)**:
+
+  - **And you're using an Amazon Elastic Container Service (ECS), Kubernetes, or Docker agent**, your code must be packaged into a Docker image and pushed to a registry your agent can access. Dagster Cloud doesn't need access to your image - your agent only needs to be able to pull it.
+
+    Additionally, the Dockerfile for your image doesn't need to specify an entry point or command. These will be supplied by the agent when it runs your code using your supplied image.
+
+  - **And you're using a local agent**, your code must be in a Python environment that can be accessed on the same machine as your agent.
+
+Additionally, note that:
+
+- Your code doesn't need to use the same version of Dagster as your agent
+- Different code locations can use different versions of Dagster
+
+---
+
+## Managing code locations in Dagster Cloud
+
+<Note>
+  <strong>Editor</strong>, <strong>Admin</strong>, or{" "}
+  <strong>Organization Admin</strong> permissions are required to manage code
+  locations in Dagster Cloud.
+</Note>
+
+- [Adding code locations](#adding-code-locations)
+- [Modifying code locations](#modifying-code-locations)
+- [Redeploying code locations](#redeploying-code-locations)
+- [Deleting code locations](#deleting-code-locations)
+
+### Adding code locations
+
+1. Sign in to your Dagster Cloud account.
+
+2. Click **Workspace**.
+
+3. Click **+ Add code location**. This will open a YAML editor with a schema describing the acceptable fields:
+
+   <Image
+   alt="Add Code Location Config Editor"
+   src="/images/dagster-cloud/developing-testing/code-locations/add_location_dagit_config.png"
+   width={1118}
+   height={511}
+   />
+
+4. In the editor, define the code location's configuration:
+
+   - `code_source` - Set this key to either `python_file:` or `package_name:` to specify where to find your code
+
+   - `executable_path` **Optional**. Define a specific Python executable if your code should run in a certain Python environment.
+
+     If not set, the code will run using the default `dagster` command-line entry-point.
+
+   - `image` - **Required if not using a local agent.** Specifies a Docker image for use with containerized agents.
+
+   - `working_directory` - Specifies the directory to use to resolve relative Python imports while loading your code.
+
+   - `attribute` - Specifies only a specific Dagster repository should be loaded
```

Comment:
nit missing period?

Reply:
we could link to Repository in the OSS docs here since repository is a pretty overloaded term